MNT: Add OneLOop in C++ example

matthewfeickert opened the pull request:
* Add example_cpp example as an additional test.
* Bump build number.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matthewfeickert commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

matthewfeickert commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13480870833](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13480870833). Examine the logs at this URL for more detail.</sub>

matthewfeickert committed:
MNT: Add OneLOop in C++ example

* Add example_cpp example as an additional test.
   - Add example_cpp/ source directory.
* Bump build number.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

matthewfeickert committed:
check env

matthewfeickert committed:
Explicitly link quadmath

matthewfeickert commented:
Here's a Pixi workspace showing things should work

```toml
[workspace]
channels = ["conda-forge"]
name = "debug-oneloop"
platforms = ["linux-64", "osx-arm64"]
version = "0.1.0"

[tasks.download]
cmd = "rm -rf oneloop && git clone https://bitbucket.org/hameren/oneloop.git"

[tasks.show]
cmd = """
pixi list oneloop && \
find $CONDA_PREFIX -type f -iname "libavh_olo*"
"""

[tasks.example_cpp]
cmd = """
cd oneloop/example_cpp/ && \
rm -f example_cpp && \
env | grep FLAGS && \
$CXX example.cpp -o example_cpp $CXXFLAGS -I. $LDFLAGS -lavh_olo -lgfortran && \
./example_cpp
"""

[tasks.debug]
cmd = """
cd src/example_cpp/ && \
rm -f example_cpp && \
$CXX example.cpp -o example_cpp $CXXFLAGS -I. $LDFLAGS -lavh_olo -lgfortran && \
./example_cpp
"""

[tasks.start]
depends-on = ["download", "show", "example_cpp"]

[dependencies]
cxx-compiler = ">=1.11.0,<2"
fortran-compiler = ">=1.11.0,<2"
oneloop = ">=3.7.2,<4"
gdb = ">=16.3,<17"
git = ">=2.51.0,<3"

[feature.static.dependencies]
cxx-compiler = ">=1.11.0,<2"
fortran-compiler = ">=1.11.0,<2"
oneloop-static = ">=3.7.2,<4"

[feature.static.tasks.show-static]
cmd = """
pixi list oneloop && \
find $CONDA_PREFIX -type f -iname "libavh_olo*"
"""

[feature.static.tasks.example_cpp-static]
cmd = """
cd oneloop/example_cpp/ && \
rm -f example_cpp && \
env | grep FLAGS && \
$CXX example.cpp -o example_cpp $CXXFLAGS -I. $LDFLAGS -lavh_olo -lgfortran -lquadmath && \
./example_cpp
"""

[tasks.start-static]
depends-on = ["show-static", "example_cpp-static"]

[environments]
static = { features = ["static"], no-default-feature = true }
```

```console
$ pixi run start
...
✨ Pixi task (show in default): pixi list oneloop && find $CONDA_PREFIX -type f -iname "libavh_olo*"                                                                                    Package  Version  Build       Size       Kind   Source
oneloop  3.7.2    h6f215e5_4  970.5 KiB  conda  https://conda.anaconda.org/conda-forge/
/tmp/debug-oneloop/.pixi/envs/default/lib/libavh_olo.so
...
########################################################################
#                                                                      #
#                        You are using OneLOop                         #
#                                                                      #
# for the evaluation of 1-loop scalar 1-, 2-, 3- and 4-point functions #
#                                                                      #
# author: Andreas van Hameren <hamerenREMOVETHIS@ifj.edu.pl>           #
#   date: 2024-11-23                                                   #
#                                                                      #
# Please cite                                                          #
#    A. van Hameren,                                                   #
#      Comput.Phys.Commun. 182 (2011) 2427-2438, arXiv:1007.4716       #
#    A. van Hameren, C.G. Papadopoulos and R. Pittau,                  #
#      JHEP 0909:106,2009, arXiv:0903.4665                             #
# in publications with results obtained with the help of this program. #
#                                                                      #
########################################################################
eps^( 0):(-1.078834873441165,0.6677066487521914)
eps^(-1):(0,0)
eps^(-2):(0,0)
```

```console
$ pixi run start-static
...
✨ Pixi task (show-static in static): pixi list oneloop && find $CONDA_PREFIX -type f -iname "libavh_olo*"                                                                              Environment: static
Package         Version  Build       Size       Kind   Source
oneloop-static  3.7.2    h6f215e5_4  989.5 KiB  conda  https://conda.anaconda.org/conda-forge/
/tmp/debug-oneloop/.pixi/envs/static/lib/libavh_olo.a
...
########################################################################
#                                                                      #
#                        You are using OneLOop                         #
#                                                                      #
# for the evaluation of 1-loop scalar 1-, 2-, 3- and 4-point functions #
#                                                                      #
# author: Andreas van Hameren <hamerenREMOVETHIS@ifj.edu.pl>           #
#   date: 2024-11-23                                                   #
#                                                                      #
# Please cite                                                          #
#    A. van Hameren,                                                   #
#      Comput.Phys.Commun. 182 (2011) 2427-2438, arXiv:1007.4716       #
#    A. van Hameren, C.G. Papadopoulos and R. Pittau,                  #
#      JHEP 0909:106,2009, arXiv:0903.4665                             #
# in publications with results obtained with the help of this program. #
#                                                                      #
########################################################################
eps^( 0):(-1.078834873441165,0.6677066487521914)
eps^(-1):(0,0)
eps^(-2):(0,0)
```

Yet in the conda-forge build (e.g. `linux-64`) things segfault

```
 │ ########################################################################
 │ #                                                                      #
 │ #                        You are using OneLOop                         #
 │ #                                                                      #
 │ # for the evaluation of 1-loop scalar 1-, 2-, 3- and 4-point functions #
 │ #                                                                      #
 │ # author: Andreas van Hameren <hamerenREMOVETHIS@ifj.edu.pl>           #
 │ #   date: 2024-11-23                                                   #
 │ #                                                                      #
 │ # Please cite                                                          #
 │ #    A. van Hameren,                                                   #
 │ #      Comput.Phys.Commun. 182 (2011) 2427-2438, arXiv:1007.4716       #
 │ #    A. van Hameren, C.G. Papadopoulos and R. Pittau,                  #
 │ #      JHEP 0909:106,2009, arXiv:0903.4665                             #
 │ # in publications with results obtained with the help of this program. #
 │ #                                                                      #
 │ ########################################################################
 │ eps^( 0):(-1.078834873441165,0.6677066487521918)
 │ eps^(-1):(0,0)
 │ eps^(-2):(0,0)
 │ $PREFIX/etc/conda/test-files/oneloop-static/0/conda_build.sh: line 29:  1567 Segmentation fault      (core dumped) ./example_cpp
```

**edit**: The segfault is recovered if the steps are run inside of a `pixi shell`. I think they aren't being shown with `pixi run` as the segfault kills the shell(?).

matthewfeickert commented:
This will get paused for a bit given issues upstream:

> It cannot work without using the dedicated Config. It has to do with arrays (certain Fortran arrays need to be fixed-size to work in C). Problem is, it still segfaults when I correct that, but for a different reason I don't understand yet.
> - Andreas van Hameren (OneLOop author) - 2025-09-19

matthewfeickert committed:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.18.12.55.27

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

matthewfeickert committed:
FIX: Add patch for bug
